odl/discr/partition.py

```diff
@@ -766,6 +767,132 @@ def uniform_partition_fromgrid(grid, begin=None, end=None):
+
+    Parameters
+    ----------
+    begin, end : float or array-like, optional
+        Vectors defining the begin end end points of an `IntervalProd`
+        (a rectangular box). For one-dimensional partitions, single
+        floats can be provided. ``None`` entries mean "compute the value".
+    num_nodes : int or sequence of int, optional
+        Number of nodes per axis. For 1d intervals, a single integer
+        can be specified. ``None`` entries mean "compute the value".
+    cell_sides : float or array-like, optional
+        Side length of the partition cells per axis. For 1d intervals,
+        a single integer can be specified. ``None`` entries mean
+        "compute the value".
+
+    Notes
+    -----
+    In each axis, 3 of the 4 possible parameters must be given. If all
+    four are provided, they are checked for consistency.
+
+    See also
+    --------
+    uniform_partition :
+        equivalent if ``begin``, ``end`` and ``num_nodes`` are given
+    uniform_partition_fromintv : partition an existing set
+    uniform_partition_fromgrid : use an existing grid as basis
+
+    Examples
+    --------
+    Any combination of three of the four parameters can be used for
+    creation of a partition:
+
+    >>> part = uniform_partition_options(begin=0, end=1, num_nodes=4)
+    >>> part.cell_boundary_vecs
+    (array([ 0.  ,  0.25,  0.5 ,  0.75,  1.  ]),)
+    >>> part = uniform_partition_options(begin=0, num_nodes=4, cell_sides=0.5)
+    >>> part.cell_boundary_vecs
+    (array([ 0. ,  0.5,  1. ,  1.5,  2. ]),)
+    >>> part = uniform_partition_options(end=1, num_nodes=4, cell_sides=0.5)
+    >>> part.cell_boundary_vecs
+    (array([-1. , -0.5,  0. ,  0.5,  1. ]),)
+    >>> part = uniform_partition_options(begin=0, end=1, cell_sides=0.5)
+    >>> part.cell_boundary_vecs
+    (array([ 0. ,  0.5,  1. ]),)
+
+    In higher dimensions, the parameters can be given differently in
+    each axis. Where None is given, the value will be computed:
+
+    >>> part = uniform_partition_options(begin=[0, 0], end=[1, 2],
+    ...                                  num_nodes=[4, 2])
+    >>> part.cell_boundary_vecs
+    (array([ 0.  ,  0.25,  0.5 ,  0.75,  1.  ]), array([ 0.,  1.,  2.]))
+    >>> part = uniform_partition_options(begin=[0, 0], end=[1, 2],
+    ...                                  num_nodes=[None, 2],
+    ...                                  cell_sides=[0.5, None])
+    >>> part.cell_boundary_vecs
+    (array([ 0. ,  0.5,  1. ]), array([ 0.,  1.,  2.]))
+    >>> part = uniform_partition_options(begin=[0, None], end=[None, 2],
+    ...                                  num_nodes=[2, 2],
+    ...                                  cell_sides=[0.5, 0.25])
+    >>> part.cell_boundary_vecs
+    (array([ 0. ,  0.5,  1. ]), array([ 1.5 ,  1.75,  2.  ]))
+    """
+    # np.size(None) = 1
+    sizes = [np.size(p) for p in (begin, end, num_nodes, cell_sides)]
+    max_size = int(np.max(sizes))
+
+    if max_size == 1:
+        begin = [begin]
```

Comment:
> This could perhaps be done on a by parameter basis:
> 
> `uniform_partition_options(begin=0, end=1, num_nodes=[1, 2, 3, 4])

What exactly do you mean by this?